src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -375,11 +385,14 @@ public override object Clone()
             }
 
             // add to known inventory
-            await this.manager.MempoolLock.WriteAsync(() =>
+            lock (this.lockObject)
             {
                 foreach (InventoryVector inventoryVector in send.Inventory)
-                    this.filterInventoryKnown.TryAdd(inventoryVector.Hash, inventoryVector.Hash);
-            });
+                {
+                    this.filterInventoryKnown.Add(inventoryVector.Hash);
+                    this.logger.LogTrace("Added inventory transaction ID '{0}' to known inventory filter.", inventoryVector.Hash);
```

Comment:
Try to avoid tracing inside a lock, is this trace really needed?

Reply:
I'll see if I can do some tracing outside so as to not slow down the lock.

Reply:
I don't think this is a good advice. It is perfectly fine to have the logger inside the lock because it cost you exactly nothing when the logging is not enabled.

Reply:
It's still bad practice and you should avoid it if you can, especially when the lock is used for quick dictionary access.